### PR TITLE
ISSUE-74 # Fix report by corr-id not by time-stamp

### PR DESCRIPTION
**Pull request by officiallysameer**

### CODE REVIEW

File: src/main/java/org/jsmart/zerocode/core/runner/ZeroCodeMultiStepsScenarioRunnerImpl.java [hunk 1]
```diff
@@ -282,7 +282,7 @@ public boolean runScenario(ScenarioSpec scenario, RunNotifier notifier, Descript
                          */
                         if(!stepOutcome) {
                             reportBuilder.result(reportResultBuilder.build());
-                            reportBuilder.printToFile(scenario.getScenarioName() + UUID.randomUUID() + ".json");
+                            reportBuilder.printToFile(scenario.getScenarioName() + logCorrelationshipPrinter.getCorrelationId() + ".json");
```
Comment (authorjapps) [inline src/main/java/org/jsmart/zerocode/core/runner/ZeroCodeMultiStepsScenarioRunnerImpl.java]: can this be unit tested?

File: src/main/java/org/jsmart/zerocode/core/logbuilder/LogCorrelationshipPrinter.java [hunk 2]
```diff
@@ -62,16 +62,17 @@ public ZeroCodeReportStep buildReportSingleStep() {
...
-                .operation(requestLogBuilder.method)
-                .requestTimeStamp(requestLogBuilder.requestTimeStamp)
+                .url(requestLogBuilder.getUrl())
+                .operation(requestLogBuilder.getMethod())
+                .requestTimeStamp(requestLogBuilder.getRequestTimeStamp())
                 .responseTimeStamp(responseLogBuilder.responseTimeStamp)
                 .responseDelay(responseDelay)
                 .build();
 
+        System.out.println("step report ---------------->>" + zeroCodeReportStep);
```
Comment (authorjapps) [inline src/main/java/org/jsmart/zerocode/core/logbuilder/LogCorrelationshipPrinter.java]: can you remove this SOUT please?

File: src/main/java/org/jsmart/zerocode/core/domain/reports/ZeroCodeReport.java [hunk 3]
```diff
@@ -15,7 +16,7 @@
     @JsonSerialize(using = LocalDateTimeSerializer.class)
```
Comment (authorjapps) [inline src/main/java/org/jsmart/zerocode/core/domain/reports/ZeroCodeReport.java]: can you take care unto nano-sec?

File: src/main/java/org/jsmart/zerocode/core/runner/ZeroCodeMultiStepsScenarioRunnerImpl.java (hunk 1 above)
Comment (officiallysameer) [reply src/main/java/org/jsmart/zerocode/core/runner/ZeroCodeMultiStepsScenarioRunnerImpl.java]: done :+1:

File: src/main/java/org/jsmart/zerocode/core/logbuilder/LogCorrelationshipPrinter.java (hunk 2 above)
Comment (officiallysameer) [reply src/main/java/org/jsmart/zerocode/core/logbuilder/LogCorrelationshipPrinter.java]: done :+1:

File: src/main/java/org/jsmart/zerocode/core/domain/reports/ZeroCodeReport.java (hunk 3 above)
Comment (officiallysameer) [reply src/main/java/org/jsmart/zerocode/core/domain/reports/ZeroCodeReport.java]: will be done as a new story #76

**Pull request merged**